File: cs/XDust/XDust/XDustExistsNode.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace XDust
{
    public class XDustExistsNode : XDustLogicNode
    {
        public XDustExistsNode(String path, String scope, Dictionary<String, XDustNode> parameters)
            : base(path, scope, parameters)
        {
        }

        public override string Operator
        {
            get
            {
                return "?";
            }
        }

        public override bool AllowIteration
        {
            get
            {
                return false;
            }
        }

        public override Context PrepareModel(XDust dust, RenderChain chain, Context context, object model)
        {
            return new Context(context, context, this.Parameters.ToDictionary(kvp => kvp.Key, kvp => kvp.Value as Object));
        }
    }
}

File: java/src/xdust/src/com/heydanno/xdust/XDustExistsNode.java
package com.heydanno.xdust;

import java.util.Map;

public class XDustExistsNode extends XDustLogicNode {

	private static final long serialVersionUID = 4129001443943748282L;

	public XDustExistsNode(String path, String scope,
			Map<String, XDustNode> parameters) {
		super(path, scope, parameters);
	}

	@Override
	public String getOperator() {
		return "?";
	}

	@Override
	public boolean getAllowIteration() {
		return false;
	}

	@Override
	public Context prepareModel(XDust dust, RenderChain chain, Context context,
			Object model) {
		return new Context(context, context, this.getParameters());
	}

}
